### PR TITLE
IGNITE-21167 Sql. Jdbc. Raise a clearer error message when the tx control statement is used in non-autocommit mode

### PR DESCRIPTION
**Pull request by xtern**
https://issues.apache.org/jira/browse/IGNITE-21167

In jdbc handler exception message has changed from
`Transaction control statement cannot be executed within an external transaction`
to more understandable
`Transaction control statements are not supported when autocommit mode is disabled`

`JdbcHandlerBase` abstract class was introduced to handle such errors uniformly.

### CODE REVIEW

**Review by korlov42 — APPROVED** [review]

File: modules/jdbc/src/integrationTest/java/org/apache/ignite/jdbc/ItJdbcMultiStatementSelfTest.java [hunk 1]
```diff
@@ -422,11 +408,18 @@ public void testAutoCommitFalse() throws Exception {
...
+    @SuppressWarnings("ThrowableNotThrown")
+    public void testAutoCommitFalseTxControlStatementsNotSupported() throws Exception {
+        String txErrMsg = "Transaction control statements are not supported when autocommit mode is disabled";
         conn.setAutoCommit(false);
-        assertThrowsSqlException(txErrMsg, () -> stmt.execute("START TRANSACTION; SELECT 1; COMMIT;"));
+        assertThrowsSqlException(txErrMsg, () -> stmt.execute("START TRANSACTION; SELECT 1; COMMIT"));
+        assertThrowsSqlException(txErrMsg, () -> stmt.execute("COMMIT"));
+        assertThrowsSqlException(txErrMsg, () -> stmt.execute("START TRANSACTION"));
+
+        boolean res = stmt.execute("SELECT 1;COMMIT");
```
Comment (zstan) [inline modules/jdbc/src/integrationTest/java/org/apache/ignite/jdbc/ItJdbcMultiStatementSelfTest.java:419]: seems we already have such a tests, it need to be merged or compound somehow 
testAutoCommitFalseNonCompleted

Comment (xtern) [reply modules/jdbc/src/integrationTest/java/org/apache/ignite/jdbc/ItJdbcMultiStatementSelfTest.java:419]: Yep,  I've just renamed it and refactored a bit

File: modules/client-handler/src/main/java/org/apache/ignite/client/handler/JdbcHandlerBase.java [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+import org.apache.ignite.internal.util.ExceptionUtils;
+import org.apache.ignite.sql.ColumnMetadata;
+import org.apache.ignite.sql.ColumnType;
+import org.apache.ignite.sql.ResultSetMetadata;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Contains common methods used to process jdbc requests.
+ */
+abstract class JdbcHandlerBase {
```
Comment (zstan) [inline modules/client-handler/src/main/java/org/apache/ignite/client/handler/JdbcHandlerBase.java:45]: abstract class with no abstract methods ?

**Review by zstan — APPROVED** [review]
overall looks good, besides one issue, a bit was discussed in private

File: modules/client-handler/src/main/java/org/apache/ignite/client/handler/JdbcHandlerBase.java (hunk 2 above)
Comment (xtern) [reply modules/client-handler/src/main/java/org/apache/ignite/client/handler/JdbcHandlerBase.java:45]: I don't see anything special here) For now abstract is used only to mark that the class cannot be instantiated on its own.

**Pull request merged**